src/shared/openssl-util.c

```diff
@@ -237,6 +163,90 @@ int pubkey_fingerprint(EVP_PKEY *pk, const EVP_MD *md, void **ret, size_t *ret_s
 
         *ret = TAKE_PTR(h);
         *ret_size = msz;
+        return 0;
+}
+
+static int pkey_generate_ec_key(int nid, EVP_PKEY **ret_ppkey) {
+
+        _cleanup_(EVP_PKEY_CTX_freep) EVP_PKEY_CTX *pctx = NULL;
+        _cleanup_(EVP_PKEY_CTX_freep) EVP_PKEY_CTX *kctx = NULL;
+        _cleanup_(EVP_PKEY_freep) EVP_PKEY *params = NULL;
+        _cleanup_(EVP_PKEY_freep) EVP_PKEY *ppkey = NULL;
+        int r;
+
+        assert(ret_ppkey);
+        assert_se(nid != NID_undef);
+
+        pctx = EVP_PKEY_CTX_new_id(EVP_PKEY_EC, NULL);
+        if (pctx)
+                return log_debug_errno(SYNTHETIC_ERRNO(EIO), "Failed to allocate pkey context");
```

Comment:
here you log at debug level. the RSA version logs at LOG_ERR level. please make this systematic, i.e. stick to LOG_ERR here too

Reply:
RSA helpers also log at debug level (for example, `rsa_encrypt_bytes` and `rsa_pkey_to_suitable_key_size`). The same scheme is used for EC helpers: main functions log at error level and helper functions log at debug level. Not sure what do you want to change here.